python/smqtk/utils/postgres.py

```diff
@@ -87,18 +91,32 @@ def __init__(self, db_name='postgres', db_host=None, db_port=None,
         self.table_upsert_lock = table_upsert_lock
         self.table_upsert_sql = None
 
+        key_tuple = (
+            self.db_name,
+            self.db_user,
+            self.db_pass,
+            self.db_host,
+            self.db_port
+        )
+        try:
+            self.connection_pool = _connection_pools[key_tuple]
+        except KeyError:
+            _connection_pools[key_tuple] = self.connection_pool = \
```

Comment:
Might want to put this in a ``threading.RLock`` in case this is performed in a threaded environment.